rocks/RocksDatabase.java

```diff
@@ -103,8 +128,18 @@ protected RocksDatabase(RocksTypeDB typedb, String name, Factory.Session session
         try {
             String schemaDirPath = directory().resolve(Encoding.ROCKS_SCHEMA).toString();
             String dataDirPath = directory().resolve(Encoding.ROCKS_DATA).toString();
-            rocksSchema = OptimisticTransactionDB.open(this.typedb.rocksDBOptions(), schemaDirPath);
-            rocksData = OptimisticTransactionDB.open(this.typedb.rocksDBOptions(), dataDirPath);
+            rocksSchemaOptions = rocksSchemaOptions();
+            rocksSchema = OptimisticTransactionDB.open(rocksSchemaOptions, schemaDirPath);
+            rocksDataOptions = rocksDataOptions();
+            if (LOG.isDebugEnabled()) {
+                rocksDataOptions.setStatistics(new Statistics()).setStatsDumpPeriodSec(ROCKS_LOG_PERIOD);
+                rocksPropertiesLogger = Executors.newScheduledThreadPool(1);
+                rocksData = OptimisticTransactionDB.open(rocksDataOptions, dataDirPath);
+                rocksPropertiesLogger.scheduleAtFixedRate(new RocksProperties.Logger(rocksData, name), 0, ROCKS_LOG_PERIOD, SECONDS);
+            } else {
+                rocksData = OptimisticTransactionDB.open(rocksDataOptions, dataDirPath);
+                rocksPropertiesLogger = null;
+            }
```

Comment:
we construct the `Options` in each Database object separately now. This is because if we share one `Options` globally, they actually end up sharing the block cache, which can have detrimental performance effects (one cache per database is best)